libgalois/include/katana/PODVector.h

```diff
@@ -163,8 +172,11 @@ class PODVector {
       return;
     }
 
-    // When reallocing, don't pay for elements greater than size_
-    shrink_to_fit();
+    // The price of unpinning&pinning again exceeds the savings below
+    if (bha_->GetType() == MemoryPinType::Swappable) {
+      // When reallocing, don't pay for elements greater than size_
+      shrink_to_fit();
+    }
```

Comment:
This may be true. But this is a subtle and surprising change.

Also tying this to pinning is not a great idea. What about a concept of a "slow" or "fast" allocator. For slow allocators, we assume that an allocation is more expensive than copying the whole buffer. For fast allocators, we assume reallocating might be better than copying the whole buffer.